src/components/document/document-workspace.tsx

```diff
@@ -207,6 +210,30 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps, {}> {
     }
   }
 
+  private handleDeleteDocument = (document: DocumentModelType) => {
+    this.stores.ui.confirm(`Delete this workspace?`, `Delete Workspace`)
+      .then((deleted: boolean) => {
+        const docType = document.type;
+        const invalidDocs = ProblemDocument || PublicationDocument || PersonalPublication ||
+                            LearningLogPublication || SupportPublication;
+        if (docType !== invalidDocs) {
+          document.setProperty("softDelete", "true");
+          this.handleDeleteOpenPrimaryDocument(deleted);
+        }
+      });
+  }
+
+  private handleDeleteOpenPrimaryDocument = async (softDelete: boolean) => {
+    const { appConfig: { defaultDocumentType, defaultDocumentContent },
+            db, ui: { problemWorkspace } } = this.stores;
+    const defaultDocument = await db.guaranteeOpenDefaultDocument(defaultDocumentType, defaultDocumentContent);
```

Comment:
As written, this won't work in Dataflow where there are no problem documents and the user is deleting the first/default personal document. I think we need to extend `guaranteeOpenDefaultDocument()` to not return deleted documents.

Reply:
I think this should maybe be a separate small PR for extending `guaranteeOpenDefaultDocument()`?